versions/2.0.0/asyncapi.md

```diff
@@ -1961,7 +1961,7 @@ Defines a security scheme that can be used by the operations. Supported schemes
 ##### Fixed Fields
 Field Name | Type | Applies To | Description
 ---|:---:|---|---
-<a name="securitySchemeObjectType"></a>type | `string` | Any | **REQUIRED**. The type of the security scheme. Valid values are `"userPassword"`, `"apiKey"`, `"X509"`, `"symmetricEncryption"`, `"asymmetricEncryption"`, `"httpApiKey"`, `"http"`, `oauth2`, and `openIdConnect`.
+<a name="securitySchemeObjectType"></a>type | `string` | Any | **REQUIRED**. The type of the security scheme. Valid values are `"userPassword"`, `"apiKey"`, `"X509"`, `"symmetricEncryption"`, `"asymmetricEncryption"`, `"httpApiKey"`, `"http"`, `"oauth2"`, and `"openIdConnect"`, `"plain"`, `"scramSha256"`, `"scramSha512"`, `"gssapi"`.
```

Comment:
Wouldn't `plain` be covered already by the `userPassword`?

Reply:
Would it make sense to reuse the `userPassword` schema, adding an optional field for the encryption?

Reply:
I think that depends whether we want `userPassword` and `X509` to be reusable across protocol.

i.e it's inferred (or we state) - that for `kafka`, `userPassword` = SASL plain, `X509` = mutual TLS, `scramShaX` = SASL scram etc

Reply:
I guess it becomes a challenge if kafka (or other protocols) add support for clashing ideas (e.g SASL x509 + mutual tls)

Reply:
> adding an optional field for the encryption
Can we assume that protocol `kafka-secure` means encrypted and `kafka` is plain?

Reply:
> i.e it's inferred (or we state) - that for `kafka`, `userPassword` = SASL plain, `X509` = mutual TLS, `scramShaX` = SASL scram etc

Perhaps it makes sense to add the concept of aliases that at the end are just reusing the same JSON Schema types?  What I would like to avoid is redefining the same concept with a different name and maintain both at the same time.

> Can we assume that protocol kafka-secure means encrypted and kafka is plain?

Yeah, that's a good point 👍 . What does it **should** mean `-secure` suffix here? I guess the question is: Do we talk about encryption or the protocol?

Reply:
> What does it should mean -secure suffix here? I guess the question is: Do we talk about encryption or the protocol?

If we compare `http/https` - the `s` represents `secure` protocol (via SSL encryption) - at least, I believe that's the case. If so, then that maps accordingly to `kafka/kafka-secure` - the `secure` means it's a secure version of the protocol?

Reply:
I agree with this. I think being consistent with the distinction between http and https would be best - so in the same way that using https means encryption and does not require authentication, I think kafka-secure should mean encryption but not require auth.

Reply:
> Would it make sense to reuse the `userPassword` schema, adding an optional field for the encryption? 

I don't really mind. 

The root requirement is that there at least three SASL mechanisms typically used for Kafka that could all be described as username/password-based (SASL/PLAIN, SASL/SCRAM-SHA-256, SASL/SCRAM-SHA-512) so we either need to describe them as different security scheme types, or make this SASL mechanism information a new additional attribute of the security scheme. 

Either is totally fine with me, as long as we have a way to articulate the required config. 

@fmvilas Any strong preferences?

Reply:
Not really. Either way is fine for me. Like you said, as long as we have a way to articulate the config, everything is fine. Just let's make sure it's not too verbose and it's maintainable in the long-term.